android-sdk/android/src/main/kotlin/io/logto/sdk/android/auth/session/LogtoAuthSession.kt

```diff
@@ -1,29 +1,39 @@
 package io.logto.sdk.android.auth.session
 
 import android.app.Activity
-import io.logto.sdk.android.auth.activity.SignInActivity
+import android.net.Uri
+import io.logto.sdk.android.auth.LogtoAuthManager
+import io.logto.sdk.android.auth.webview.WebViewAuthActivity
+import io.logto.sdk.android.completion.Completion
 import io.logto.sdk.android.exception.LogtoException
 import io.logto.sdk.android.type.LogtoConfig
 import io.logto.sdk.core.Core
 import io.logto.sdk.core.exception.CallbackUriVerificationException
-import io.logto.sdk.core.http.HttpCompletion
 import io.logto.sdk.core.type.CodeTokenResponse
 import io.logto.sdk.core.type.OidcConfigResponse
 import io.logto.sdk.core.util.CallbackUriUtils
 import io.logto.sdk.core.util.GenerateUtils
 
-class SignInSession(
+class LogtoAuthSession(
     val context: Activity,
     val logtoConfig: LogtoConfig,
     val oidcConfig: OidcConfigResponse,
     val redirectUri: String,
-    val completion: HttpCompletion<CodeTokenResponse>,
-) {
+    override val completion: Completion<CodeTokenResponse>,
+) : AuthSession<CodeTokenResponse> {
+
     private val codeVerifier = GenerateUtils.generateCodeVerifier()
     private val state = GenerateUtils.generateState()
 
-    fun start() {
-        SignInSessionManager.setSession(this)
+    override fun start() {
+        val redirectUriScheme = Uri.parse(redirectUri).scheme
+        if (redirectUriScheme == null) {
+            completion.onComplete(LogtoException(LogtoException.Message.INVALID_REDIRECT_URI), null)
+            return
+        }
+
+        LogtoAuthManager.handleAuthStart(redirectUriScheme, this)
```

Comment:
why do we need this? we can just pass completion inside activity?

Reply:
We have to handle the session across the `CallbackUriActivity` and the `WebViewAuthActivity'. So we need a `Manager` to handle this.

Reply:
yes we can just pass callback into another activity?

Reply:
No, we can't.
https://stackoverflow.com/questions/2091465/how-do-i-pass-data-between-activities-in-android-application